packages/sit-onyx/src/utils/comparator.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { expect, test } from "vitest";
+import { areObjectsFlatEqual } from "./comparator";
+
+const referenceObj = { a: 42, b: "foo", c: null, d: true };
+
+test("should return true for equal objects with the same ordered key-values", () => {
+  // ARRANGE
+  const compareObj = { ...referenceObj };
+
+  // ACT
+  const result = areObjectsFlatEqual(referenceObj, compareObj);
+
+  // ASSERT
+  expect(result).toBeTruthy();
+});
+
+test("should return true for equal objects with differently ordered key-values", () => {
+  // ARRANGE
+  const compareObj = { d: true, b: "foo", c: null, a: 42 };
+
+  // ACT
+  const result = areObjectsFlatEqual(referenceObj, compareObj);
+
+  // ASSERT
+  expect(result).toBeTruthy();
+});
+
+test("should return true for equal objects ignoring undefined keys", () => {
+  // ARRANGE
+  const compareObj = { ...referenceObj, e: undefined };
+
+  // ACT
+  const result = areObjectsFlatEqual(referenceObj, compareObj);
+
+  // ASSERT
+  expect(result).toBeTruthy();
+});
+
+test.each([
```

Comment:
(optional) You could also combine all the test cases above into this test.each by adding a second property `shouldBeEqual`

Reply:
I'll try, I just hope it does not make the test label too hard to build :D

Reply:
![image](https://github.com/SchwarzIT/onyx/assets/151019977/b04693d5-3908-465f-95d5-bc738fe62a1b)
the test message gets truncated :'(

Reply:
![image](https://github.com/SchwarzIT/onyx/assets/151019977/dda40b1d-e767-4b58-af09-1f1322eedcf7)
even if I make them fail. imho it's more important that the messages are complete then to get all in one test.each.

but I'll bundle up the truthy tests, better than nothing ^^